src/merkle/tiered_smt/mod.rs

```diff
@@ -480,10 +490,13 @@ fn get_common_prefix_tier_depth(key1: &RpoDigest, key2: &RpoDigest) -> u8 {
 /// Computes node value for leaves at tiers 16, 32, or 48.
 ///
 /// Node value is computed as: hash(key || value, domain = depth).
-pub fn hash_upper_leaf(key: RpoDigest, value: Word, depth: u8) -> RpoDigest {
-    const NUM_UPPER_TIERS: usize = TieredSmt::TIER_DEPTHS.len() - 1;
-    debug_assert!(TieredSmt::TIER_DEPTHS[..NUM_UPPER_TIERS].contains(&depth));
-    Rpo256::merge_in_domain(&[key, value.into()], depth.into())
+pub fn hash_upper_leaf<T>(key: RpoDigest, value: &T, depth: u8) -> RpoDigest
+where
+    T: Copy + Into<RpoDigest>,
+{
+    const NUM_UPPER_TIERS: usize = TIER_DEPTHS.len() - 1;
+    debug_assert!(TIER_DEPTHS[..NUM_UPPER_TIERS].contains(&depth));
+    Rpo256::merge_in_domain(&[key, (*value).into()], depth.into())
```

Comment:
This is a subtle breaking change.

Before `value.into()` was converting `Word` into `&[u8]`. Now this hashes the value instead.